File: turbo/benchmarks/c/collatz.c
#include <stdio.h>

long collatz_len(long n) {
    long x = n, l = 0;
    while (x != 1) {
        if (x % 2 == 0) x = x / 2; else x = 3 * x + 1;
        l++;
    }
    return l;
}

int main() {
    long max_len = 0;
    for (long i = 1; i <= 1000000; i++) {
        long l = collatz_len(i);
        if (l > max_len) max_len = l;
    }
    printf("%ld\n", max_len);
    return 0;
}

File: turbo/benchmarks/rust/collatz.rs
fn collatz_len(n: i64) -> i64 {
    let (mut x, mut l) = (n, 0i64);
    while x != 1 {
        x = if x % 2 == 0 { x / 2 } else { 3 * x + 1 };
        l += 1;
    }
    l
}
fn main() {
    let mut max_len = 0i64;
    for i in 1..=1_000_000i64 {
        let l = collatz_len(i);
        if l > max_len { max_len = l; }
    }
    println!("{}", max_len);
}
